src/Gnosis.sol

```diff
@@ -21,25 +20,65 @@ library Gnosis {
     address internal constant WETH_GATEWAY                     = 0xBD7D6a9ad7865463DE44B05F04559f65e3B11704;
 
     /******************************************************************************************************************/
-    /*** Data Provider Addresses                                                                                    ***/
+    /*** Reserve Token Addresses                                                                                    ***/
     /******************************************************************************************************************/
 
-    address internal constant PROTOCOL_DATA_PROVIDER     = 0x2a002054A06546bB5a264D57A81347e23Af91D18;
-    address internal constant UI_INCENTIVE_DATA_PROVIDER = 0xA7F8A757C4f7696c015B595F51B2901AC0121B18;
-    address internal constant UI_POOL_DATA_PROVIDER      = 0xF028c2F4b19898718fD0F77b9b881CbfdAa5e8Bb;
-    address internal constant WALLET_BALANCE_PROVIDER    = 0xd2AeF86F51F92E8e49F42454c287AE4879D1BeDc;
+    address internal constant GNO                   = 0x9C58BAcC331c9aa871AFD802DB6379a98e80CEdb;
+    address internal constant GNO_ATOKEN            = 0x2f589BADbE2024a94f144ef24344aF91dE21a33c;
+    address internal constant GNO_STABLE_DEBT_TOKEN = 0xd4bAbF714964E399f95A7bb94B3DeaF22d9F575d;
+    address internal constant GNO_DEBT_TOKEN        = 0xE74569197bFba0e2231c84587cc5f9ceDB807043;
```

Comment:
These are off by one again. Addresses are: `0x5671b0B8aC13DC7813D36B99C21c53F6cd376a14`, `0x2f589BADbE2024a94f144ef24344aF91dE21a33c`  and `0xd4bAbF714964E399f95A7bb94B3DeaF22d9F575d`.

Reply:
fixed